wrapper/src/main/java/software/amazon/jdbc/plugin/bluegreen/BlueGreenStatusProvider.java

```diff
@@ -509,55 +521,26 @@ protected void updateDnsFlags(BlueGreenRole role, BlueGreenInterimStatus interim
       LOGGER.finest(() -> Messages.get("bgd.greenTopologyChanged", new Object[] {this.bgdId}));
       this.greenTopologyChanged = true;
       this.storeGreenTopologyChangeTime();
+      this.resetMonitors(this.monitorResetOnTopologyCompleted, "- green topology");
     }
   }
 
-  protected int getInterimStatusHash(BlueGreenInterimStatus interimStatus) {
-
-    int result = this.getValueHash(1,
-        interimStatus.blueGreenPhase == null ? "" : interimStatus.blueGreenPhase.toString());
-    result = this.getValueHash(result,
-        interimStatus.version == null ? "" : interimStatus.version);
-    result = this.getValueHash(result, String.valueOf(interimStatus.port));
-    result = this.getValueHash(result, String.valueOf(interimStatus.allStartTopologyIpChanged));
-    result = this.getValueHash(result, String.valueOf(interimStatus.allStartTopologyEndpointsRemoved));
-    result = this.getValueHash(result, String.valueOf(interimStatus.allTopologyChanged));
-
-    result = this.getValueHash(result,
-        interimStatus.hostNames == null
-            ? ""
-            : interimStatus.hostNames.stream()
-                .sorted(Comparator.comparing(x -> x))
-                .collect(Collectors.joining(",")));
-    result = this.getValueHash(result,
-        interimStatus.startTopology == null
-            ? ""
-            : interimStatus.startTopology.stream()
-                .map(x -> x.getHostAndPort() + x.getRole())
-                .sorted(Comparator.comparing(x -> x))
-                .collect(Collectors.joining(",")));
-    result = this.getValueHash(result,
-        interimStatus.currentTopology == null
-            ? ""
-            : interimStatus.currentTopology.stream()
-                .map(x -> x.getHostAndPort() + x.getRole())
-                .sorted(Comparator.comparing(x -> x))
-                .collect(Collectors.joining(",")));
-    result = this.getValueHash(result,
-        interimStatus.startIpAddressesByHostMap == null
-            ? ""
-            : interimStatus.startIpAddressesByHostMap.entrySet().stream()
-                .map(x -> x.getKey() + x.getValue())
-                .sorted(Comparator.comparing(x -> x))
-                .collect(Collectors.joining(",")));
-    result = this.getValueHash(result,
-        interimStatus.currentIpAddressesByHostMap == null
-            ? ""
-            : interimStatus.currentIpAddressesByHostMap.entrySet().stream()
-                .map(x -> x.getKey() + x.getValue())
-                .sorted(Comparator.comparing(x -> x))
-                .collect(Collectors.joining(",")));
-    return result;
+  protected void resetMonitors(final AtomicBoolean monitorResetCompleted, final String eventName) {
+    if (monitorResetCompleted.compareAndSet(false, true)) {
```

Comment:
Here we are setting monitorResetCompleted before the reset is actually performed below, should we rename to monitorResetRequested, or check the value here but update it after the block below is finished?

Reply:
`monitorResetCompleted` is a flag that helps to reset monitor just once. I think it doesn't matter where to set it up, before actual reset or after.